### PR TITLE
Create Parsons Mailchimp connector

### PR DESCRIPTION
**Pull request by SorenSpicknall**

### CODE REVIEW

File: parsons/mailchimp/mailchimp.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    `Args:`
+        api_key:
+            The Mailchimp-provided application key. Not required if
+            ``MAILCHIMP_API_KEY`` env variable set.
+    `Returns:`
+        Mailchimp Class
+    """
+
+    def __init__(self, api_key):
+        self.api_key = api_key
```
Comment (jburchard) [inline parsons/mailchimp/mailchimp.py:22]: Stealing from Eliot's comments on another PR:

When possible, we want Parsons connectors to support configuration via environment variables as well as through the arguments of the `__init__` method. The way this is usually done is by making the arguments optional (ie `api_token=None`) and then using the `parsons.check_env.check` method to try to pull the value from the environment. As an example, see the Rock the Vote connector: https://github.com/move-coop/parsons/blob/14e369ec8aa325bed054fb859fa4a2e8789b2aef/parsons/rockthevote/rtv.py#L48-L50

File: parsons/mailchimp/mailchimp.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+    def get_members(self, list_id, fields=None,
+                    exclude_fields=None, count=None, offset=None,
+                    email_type=None, status=None, since_timestamp_opt=None,
+                    before_timestamp_opt=None, since_last_changed=None,
+                    before_last_changed=None, unique_email_id=None,
+                    vip_only=False, interest_category_id=None,
+                    interest_ids=None, interest_match=None, sort_field=None,
+                    sort_dir=None, since_last_campaign=None,
+                    unsubscribed_since=None):
```
Comment (jburchard) [inline parsons/mailchimp/mailchimp.py:110]: Needs doc strings.

File: parsons/mailchimp/mailchimp.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+            return tbl
+        else:
+            return None
+
+    def get_campaigns(self, fields=None, exclude_fields=None,
+                      count=None, offset=None, type=None, status=None,
+                      before_send_time=None, since_send_time=None,
+                      before_create_time=None, since_create_time=None,
+                      list_id=None, folder_id=None, member_id=None,
+                      sort_field=None, sort_dir=None):
```
Comment (jburchard) [inline parsons/mailchimp/mailchimp.py:76]: Needs doc strings.

File: parsons/mailchimp/mailchimp.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        tbl = Table(response['members'])
+        logger.info(f'Found {tbl.num_rows} members.')
+        if tbl.num_rows > 0:
+            return tbl
+        else:
+            return None
+
+    def get_campaign_emails(self, campaign_id, fields=None,
+                            exclude_fields=None, count=None, offset=None,
+                            since=None):
```
Comment (jburchard) [inline parsons/mailchimp/mailchimp.py:143]: Needs doc strings.

File: parsons/mailchimp/mailchimp.py [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        if fields_to_expand:
+            [tbl.unpack_dict(x, prepend=False) for x in fields_to_expand]
+
+        return tbl
+
+    def get_lists(self, fields=None, exclude_fields=None,
+                  count=None, offset=None, before_date_created=None,
+                  since_date_created=None, before_campaign_last_sent=None,
+                  since_campaign_last_sent=None, email=None, sort_field=None,
+                  sort_dir=None):
```
Comment (jburchard) [inline parsons/mailchimp/mailchimp.py:49]: Needs doc strings.

File: parsons/mailchimp/mailchimp.py [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+                  'vip_only': vip_only,
+                  'interest_category_id': interest_category_id,
+                  'interest_ids': interest_ids,
+                  'interest_match': interest_match,
+                  'sort_field': sort_field,
+                  'sort_dir': sort_dir,
+                  'since_last_campaign': since_last_campaign,
+                  'unsubscribed_since': unsubscribed_since}
+
+        response = self.get_request('lists/' + list_id + '/members',
```
Comment (jburchard) [inline parsons/mailchimp/mailchimp.py:132]: Consider an f-string here for consistency with other classes:
```
response = self.get_request(f"lists/{list_id}/members', params=params)
```

File: parsons/mailchimp/mailchimp.py [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+
+        params = {'fields': fields,
+                  'exclude_fields': exclude_fields,
+                  'count': count,
+                  'offset': offset,
+                  'since': since}
+
+        response = self.get_request('reports/' + campaign_id +
+                                    '/email-activity',
+                                    params=params)
```
Comment (jburchard) [inline parsons/mailchimp/mailchimp.py:153]: Consider f-string. See above for example.

File: parsons/mailchimp/mailchimp.py [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+    def get_unsubscribes(self, campaign_id, fields=None,
+                         exclude_fields=None, count=None, offset=None):
+
+        params = {'fields': fields,
+                  'exclude_fields': exclude_fields,
+                  'count': count,
+                  'offset': offset}
+
+        response = self.get_request('reports/' + campaign_id + '/unsubscribed',
+                                    params=params)
```
Comment (jburchard) [inline parsons/mailchimp/mailchimp.py:169]: Consider f-string

File: parsons/mailchimp/mailchimp.py [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+                                    '/email-activity',
+                                    params=params)
+        tbl = Table(response['emails'])
+        if tbl.num_rows > 0:
+            return tbl
+        else:
+            return None
+
+    def get_unsubscribes(self, campaign_id, fields=None,
+                         exclude_fields=None, count=None, offset=None):
```
Comment (jburchard) [inline parsons/mailchimp/mailchimp.py:161]: Doc strings.

File: parsons/mailchimp/mailchimp.py (hunk 1 above)
Comment (SorenSpicknall) [reply parsons/mailchimp/mailchimp.py]: Solved in 70c6309.

File: parsons/mailchimp/mailchimp.py (hunk 3 above)
Comment (SorenSpicknall) [reply parsons/mailchimp/mailchimp.py:130]: Solved in b1d33cd.

File: parsons/mailchimp/mailchimp.py (hunk 5 above)
Comment (SorenSpicknall) [reply parsons/mailchimp/mailchimp.py:60]: Solved in b1d33cd.

File: parsons/mailchimp/mailchimp.py (hunk 4 above)
Comment (SorenSpicknall) [reply parsons/mailchimp/mailchimp.py:327]: Solved in 41330c0.

File: parsons/mailchimp/mailchimp.py (hunk 2 above)
Comment (SorenSpicknall) [reply parsons/mailchimp/mailchimp.py:216]: Solved in 41330c0.

File: parsons/mailchimp/mailchimp.py (hunk 9 above)
Comment (SorenSpicknall) [reply parsons/mailchimp/mailchimp.py:372]: Solved in 41330c0.

File: parsons/mailchimp/mailchimp.py (hunk 8 above)
Comment (SorenSpicknall) [reply parsons/mailchimp/mailchimp.py]: Solved in 835df40.

File: parsons/mailchimp/mailchimp.py (hunk 7 above)
Comment (SorenSpicknall) [reply parsons/mailchimp/mailchimp.py]: Solved in 835df40.

File: parsons/mailchimp/mailchimp.py (hunk 6 above)
Comment (SorenSpicknall) [reply parsons/mailchimp/mailchimp.py]: Solved in 835df40.

**Comment by SorenSpicknall** [comment]
I added RST documentation in 4b1a3e0, and resolved other specific feedback in additional commits. Could you expand on what you mean by this, though, @jburchard?

> mailchimp.py import top level __init__.py

**Comment by jburchard** [comment]
Sure.

Just import the class into this file:

File: parsons/mailchimp/mailchimp.py [hunk 10]
```diff
@@ -0,0 +1,391 @@
...
+                  'sort_field': sort_field,
+                  'sort_dir': sort_dir}
+
+        response = self.get_request('lists', params=params)
+        tbl = Table(response['lists'])
+        logger.info(f'Found {tbl.num_rows} lists.')
+        if tbl.num_rows > 0:
+            return tbl
+        else:
+            return None
```
Comment (eliotst) [inline parsons/mailchimp/mailchimp.py:104]: This method (and those below) should return an empty table instead of `None`.

Comment (SorenSpicknall) [reply parsons/mailchimp/mailchimp.py]: Fixed in d9e1550.

**Comment by SorenSpicknall** [comment]
I added some testing with basic assertions, which we can expand upon down the road, and took care of the remaining feedback items. Ready for re-review.

**Pull request merged**